packages/front-end/components/DataViz/DataVizDimensionPanel.tsx

```diff
@@ -0,0 +1,302 @@
+import Collapsible from "react-collapsible";
+import { Box, Flex, Separator, Text, TextField } from "@radix-ui/themes";
+import { PiNetwork, PiTrash } from "react-icons/pi";
+import { FaAngleRight, FaPlusCircle } from "react-icons/fa";
+import { DataVizConfig } from "back-end/src/validators/saved-queries";
+import { Select, SelectItem } from "@/components/Radix/Select";
+import { supportsDimension } from "@/services/dataVizTypeGuards";
+import Badge from "@/components/Radix/Badge";
+import Button from "@/components/Radix/Button";
+
+export default function DataVizDimensionPanel({
+  dataVizConfig,
+  onDataVizConfigChange,
+  axisKeys,
+}: {
+  dataVizConfig: Partial<DataVizConfig>;
+  onDataVizConfigChange: (dataVizConfig: Partial<DataVizConfig>) => void;
+  axisKeys: string[];
+}) {
+  if (!supportsDimension(dataVizConfig)) {
+    return null;
+  }
+
+  const dimensions = dataVizConfig.dimension || [];
```

Comment:
suggestion: We should explicitly cast dimensions as `dimensionAxisConfiguration[]`.

Otherwise the `|| []` removes the TS ability to know what this is and on line `94` the dimension gets treated as `any`.

Reply:
Will do.

This had a side effect of TS not being happy with setting the default value for a dimension's `fieldName` to an empty string.

As a result, I've just set the default value of the dimension to be the first index of the `axisKeys`.